dex/src/state.rs

```diff
@@ -2686,8 +2691,173 @@ impl State {
     }
 
     #[cfg(feature = "program")]
-    fn process_send_take(_args: account_parser::SendTakeArgs) -> DexResult {
-        unimplemented!()
+    fn process_send_take(args: account_parser::SendTakeArgs) -> DexResult {
+
+        let account_parser::SendTakeArgs {
+            instruction,
+            signer,
+            mut event_q,
+            mut req_q,
+            mut order_book_state,
+            coin_wallet,
+            pc_wallet,
+            coin_vault,
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer
+        } = args;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side); 
+        let native_pc_qty_locked = match instruction.side {
+            Side::Bid => {
+                let lock_qty_native = instruction.max_native_pc_qty_including_fees;
+                Some(lock_qty_native)
+            }
+            Side::Ask => {
+                None
+            }
+        };
+
+        let request = RequestView::NewOrder {
+            side: instruction.side,
+            order_type: OrderType::ImmediateOrCancel,
+            order_id,
+            fee_tier,
+            self_trade_behavior: SelfTradeBehavior::AbortTransaction,
+            // Pass dummy account - system program pubkey
+            owner: system_program::ID.to_aligned_bytes(),
+            // Pass 0 
+            owner_slot: 0,
+            max_coin_qty: instruction.max_coin_qty,
+            native_pc_qty_locked,
+            client_order_id: None,
+        };
+
+        let mut limit = instruction.limit;
```

Comment:
If this limit is zero/gets decremented to zero in `new_order` we will still post to the orderbook (line 266), but without an valid OpenOrders account. Probably would make sense to change the logic in matching.rs:new_order for this case. It also seems odd to me that the IOC Order Type doesn't have this integrated...

Reply:
Yeah I made this change in the newest update. Post allowed is no longer flipped to true